src/librustc/middle/const_eval.rs

```diff
@@ -416,6 +424,12 @@ pub enum ErrKind {
     ExpectedConstTuple,
     ExpectedConstStruct,
     TupleIndexOutOfBounds,
+    IndexedNonVec,
+    IndexNotNatural,
```

Comment:
Positive doesn't include zero. (Usually the naturals start at zero.)

You could do `IndexNotNonNegative`, or just `IndexNegative` if you like.  :)